src/integrationTest/java/org/opensearch/security/http/UntrustedLdapServerCertificateTest.java

```diff
@@ -0,0 +1,92 @@
+import org.opensearch.test.framework.cluster.TestRestClient;
+import org.opensearch.test.framework.ldap.EmbeddedLDAPServer;
+
+import static org.opensearch.security.http.DirectoryInformationTrees.DN_OPEN_SEARCH_PEOPLE_TEST_ORG;
+import static org.opensearch.security.http.DirectoryInformationTrees.DN_PEOPLE_TEST_ORG;
+import static org.opensearch.security.http.DirectoryInformationTrees.LDIF_DATA;
+import static org.opensearch.security.http.DirectoryInformationTrees.PASSWORD_OPEN_SEARCH;
+import static org.opensearch.security.http.DirectoryInformationTrees.PASSWORD_SPOCK;
+import static org.opensearch.security.http.DirectoryInformationTrees.USERNAME_ATTRIBUTE;
+import static org.opensearch.security.http.DirectoryInformationTrees.USER_SEARCH;
+import static org.opensearch.security.http.DirectoryInformationTrees.USER_SPOCK;
+import static org.opensearch.test.framework.TestSecurityConfig.AuthcDomain.AUTHC_HTTPBASIC_INTERNAL;
+import static org.opensearch.test.framework.TestSecurityConfig.Role.ALL_ACCESS;
+
+/**
+* Negative test case related to LDAP server certificate. Connection between OpenSearch and LDAP server should not be established if
+* OpenSearch "does not trust" LDAP server certificate.
+*/
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class UntrustedLdapServerCertificateTest {
+
+	private static final TestSecurityConfig.User ADMIN_USER = new TestSecurityConfig.User("admin").roles(ALL_ACCESS);
+
+	private static final TestCertificates TEST_CERTIFICATES = new TestCertificates();
+
+	public static final EmbeddedLDAPServer embeddedLDAPServer = new EmbeddedLDAPServer(TEST_CERTIFICATES.getRootCertificateData(),
+		TEST_CERTIFICATES.createSelfSignedCertificate("CN=untrusted"), LDIF_DATA);
+
+	public static LocalCluster cluster = new LocalCluster.Builder()
+		.testCertificates(TEST_CERTIFICATES)
+		.clusterManager(ClusterManager.SINGLENODE).anonymousAuth(false)
+		.authc(new AuthcDomain("ldap", 2, true)
+			.httpAuthenticator(new HttpAuthenticator("basic").challenge(false))
+			.backend(new AuthenticationBackend("ldap")
+				.config(() -> LdapAuthenticationConfigBuilder.config()
+					// this port is available when embeddedLDAPServer is already started, therefore Supplier interface is used
+					.hosts(List.of("localhost:" + embeddedLDAPServer.getLdapsPort()))
+					.enableSsl(true)
+					.bindDn(DN_OPEN_SEARCH_PEOPLE_TEST_ORG)
+					.password(PASSWORD_OPEN_SEARCH)
+					.userBase(DN_PEOPLE_TEST_ORG)
+					.userSearch(USER_SEARCH)
+					.usernameAttribute(USERNAME_ATTRIBUTE)
+					.penTrustedCasFilePath(TEST_CERTIFICATES.getRootCertificate().getAbsolutePath())
+					.build())))
+		.authc(AUTHC_HTTPBASIC_INTERNAL)
+		.users(ADMIN_USER)
+		.build();
+
+	@ClassRule
+	public static RuleChain ruleChain = RuleChain.outerRule(embeddedLDAPServer).around(cluster);
+
+	@Test
+	public void shouldAuthenticateUserWithLdap_positive() {
```

Comment:
this name is a little misleading. `shouldNotAuthenticateUserWithLdap_positive` sounds good. What do you say?

Reply:
done